### PR TITLE
Fix comment leakage in exports, false circular-reference log redaction, and API key auth parsing

### PR DESCRIPTION
**Pull request by Phazzie**
Three small, independent defects found by probing existing utility modules against the behaviour their tests already imply. Each is covered by a new regression assertion.

## 1. HTML comments leak into exported stories

`tokenizeHtml` had no case for an HTML comment, so a comment was split at the first `>` it contained and its body fell through as text:

```
input:  <p>Hi</p><!&#45;&#45; editor note &#45;&#45;><p>Bye</p>
output: <p>Hi</p>editor note &#45;&#45;&amp;gt;<p>Bye</p>
```

Comments are now consumed whole in the tokenizer, so both `sanitizeStoryHtmlForExport` and `stripStoryHtmlForExport` drop them. An unterminated comment swallows the remainder of the document (matching browser parsing) rather than leaking the markup after it.

## 2. Shared object references logged as `[Circular]`

`redactValue` added every visited object to a `WeakSet` and never removed it, so the set meant "seen anywhere" rather than "on the current path". Any object referenced twice from different branches lost its contents:

```js
const ctx = { model: 'grok-4' };
redactSensitiveLogData({ a: ctx, b: ctx })
// { a: { model: 'grok-4' }, b: '[Circular]' }
```

The set now tracks only the ancestors of the current recursion (unwound in a `finally`), so repeated references are redacted normally. Genuine cycles are still broken — asserted by a new test.

## 3. API key authentication rejected valid keys

- `API_KEYS=key1, key2` rejected `key2`: entries were tested with `k.trim()` but stored untrimmed, so the configured value was `" key2"`.
- The Bearer strip was case-sensitive and unanchored, so a lowercase `bearer` prefix failed and a key containing the literal scheme text anywhere was mangled. Replaced with an anchored, case-insensitive prefix check that requires a whitespace separator.
- Headers arriving as `string[]` (repeated headers, which Node produces) were passed straight to comparison and never matched.
- A Bearer scheme with no credentials after it now reports `MISSING_API_KEY` rather than being treated as a key named "Bearer" (review feedback).
- Key comparison moved to `timingSafeEqual` so a valid key cannot be recovered byte by byte from response timing.
- The module-scope `setInterval` for rate-limit cleanup is now unref'd. Previously importing this module kept the process alive indefinitely — reproducible today by importing it from any script, which then never exits.

## Testing

- `npm run test:all` — green, including the new `test:api-key-auth` suite wired into it.
- New assertions in `tests/export-sanitizer.test.ts` and `tests/log-redaction.test.ts`; each was confirmed to fail against the pre-fix code.
- No behavioural change to the passing paths any existing test covers.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01Eng8BahVqzuJAmB9f4SmEj

## Summary by Sourcery

Harden story exports, log redaction, and API key authentication against comment leakage, false circular-reference detection, and malformed or valid-but-unsupported header inputs.

Bug Fixes:
- Prevent HTML comments, including unterminated comments, from leaking into exported story content.
- Correct log redaction to distinguish shared references from genuine circular references.
- Improve API key authentication for trimmed configuration values, Bearer headers, repeated headers, missing credentials, and constant-time key comparison.

Enhancements:
- Allow the rate-limit cleanup timer to be unref'd so importing the security middleware does not keep processes alive.

Tests:
- Add regression coverage for export comment handling, shared and circular log references, and API key authentication edge cases.
- Include the API key authentication suite in the full test command.

### CODE REVIEW

File: api/_lib/middleware/security.ts [hunk 1]
```diff
@@ -82,9 +88,47 @@ export async function authenticateRequest(req: AuthenticatedRequest): Promise<Au
...
+ * per RFC 7235 and only counts at the start of the value.
+ */
+function stripBearerPrefix(value: string | undefined): string | undefined {
+  if (!value) {
+    return undefined;
+  }
+
+  const match = /^bearer\s+(.+)$/i.exec(value);
+  return (match ? match[1].trim() : value) || undefined;
+}
```
[the remark this answers is not in the document]
Comment (Phazzie) [reply api/_lib/middleware/security.ts]: Fixed in 3f4b8fa. `stripBearerPrefix` now returns `undefined` when the value is the scheme with no credentials after it (`Bearer`, or `Bearer` followed only by whitespace), so `authenticateRequest` reports `MISSING_API_KEY`. A value that merely starts with the scheme name — `bearerkey` — is still returned intact and treated as a key in its own right, since there is no separator to indicate a scheme. Both cases are covered by new assertions in `tests/api-key-auth.test.ts`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/api-key-auth.test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env tsx
+// Created: 2026-08-24 UTC
+
```
[the remark this answers is not in the document]
Comment (Phazzie) [reply tests/api-key-auth.test.ts]: Fixed in 3f4b8fa — header is now `// Created: 2026-08-24 13:58 UTC`, matching the `YYYY-MM-DD HH:MM TZ` form used by the other test scripts.

---
_Generated by [Claude Code](https://claude.ai/code)_
